src/server.ts

```diff
@@ -80,25 +89,15 @@ connection.onDidChangeConfiguration(params => {
 });
 
 // Update cache
-connection.onDidChangeWatchedFiles(async event => {
-	const firstEvent = event.changes[0];
-	const isSameDocumentPath = activeDocumentUri === firstEvent.uri;
-	const isRenameAction = firstEvent.type === 1;
-
-	// We do not need to update the Cache if the current document has been updated
-	// But we should warm up Cache if is rename action
-	if (event.changes.length === 1 && isSameDocumentPath && !isRenameAction) {
-		return;
-	}
+connection.onDidChangeWatchedFiles(event => {
+	const files = event.changes
+		.filter(file => file.type === FileChangeType.Changed || file.type === FileChangeType.Created)
+		.map(file => Files.uriToFilePath(file.uri));
 
-	const symbols = await doScanner(workspaceRoot, cache, settings);
-
-	invalidateCacheStorage(cache, symbols);
+	return scannerService.scan(files, /* recursive */ false);
```

Comment:
Most likely here we need to always use `recursive` mode, because a new import may appear in the document.